connector/kafka-manager/src/main/java/it/pagopa/selfcare/external_interceptor/connector/kafka_manager/factory/SendFdNotification.java

```diff
@@ -55,18 +60,51 @@ public void sendUserNotification(UserNotification userNotification, Acknowledgme
+            if (userNotification.getEventType().equals(QueueEvent.UPDATE) && userNotification.getUser().getRelationshipStatus() == null) {
+                UserProductDetails userProduct = externalApiConnector.getUserOnboardedProductDetails(userNotification.getUser().getUserId(), userNotification.getInstitutionId(), userNotification.getProductId());
+                if(userProduct.getOnboardedProductDetails() != null) {
+                    sendUpdateUserEvents(notificationToSend, userProduct);
+                    acknowledgment.acknowledge();
+                }
+            } else {
+                notificationToSend.setType(NotificationType.getNotificationTypeFromRelationshipState(userNotification.getUser().getRelationshipStatus()));
+                String userNotificationToSend = mapper.writeValueAsString(notificationToSend);
+                String topic = producerAllowedTopics.get().get(userNotification.getProductId());
+                String logSuccess = String.format("sent notification for user : %s, to FD", userNotification.getUser().getUserId());
+                String logFailure = String.format("error during notification sending for user %s: {}, on FD ", userNotification.getUser().getUserId());
+                sendNotification(userNotificationToSend, topic, logSuccess, logFailure, Optional.of(acknowledgment));
+            }
         }
         log.trace("sendUserNotification end");
 
     }
 
-    private boolean validateProductTopic(String productId){
+    private void sendUpdateUserEvents(NotificationToSend notification, UserProductDetails userProductDetails) {
+        log.trace("sendUpdateUserEvents start");
+        List<NotificationType> eventTypes = List.of(NotificationType.DELETE_USER, NotificationType.ACTIVE_USER);
+        UserToSend userToSend = new UserToSend();
+        userToSend.setUserId(userProductDetails.getId());
+        userToSend.setRole(userProductDetails.getOnboardedProductDetails().getRole());
+        userToSend.setRoles(userProductDetails.getOnboardedProductDetails().getRoles());
+        notification.setCreatedAt(userProductDetails.getOnboardedProductDetails().getCreatedAt());
+        eventTypes.forEach(type -> {
+            notification.setType(type);
+            notification.setUser(userToSend);
+            String userNotificationToSend = null;
+            try {
+                userNotificationToSend = mapper.writeValueAsString(notification);
+            } catch (JsonProcessingException e) {
+                throw new RuntimeException(e);
+            }
+            String topic = producerAllowedTopics.get().get(notification.getProduct());
+            String logSuccess = String.format("sent notification for user : %s, to FD", notification.getUser().getUserId());
+            String logFailure = String.format("error during notification sending for user %s: {}, on FD ", notification.getUser().getUserId());
```

Comment:
you could introduce a constant class variabile for the string message, for ex. MESSAGE_LOG_SUCCESS